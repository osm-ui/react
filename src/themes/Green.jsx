import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import { colors } from 'constants/index';
import { buildDarkThemeConfig } from 'helpers/themes';
import { config as defaultThemeConfig, StyledSpan } from './Default';

export const config = _.merge(
  {},
  buildDarkThemeConfig(defaultThemeConfig, colors, 'green'),
  {
    form: {
      button: {
        success: {
          color: colors.white,
          backgroundColor: colors.green3,
          borderColor: colors.green3,
          hoverBackgroundColor: colors.green1,
          hoverBorderColor: colors.green1,
          focusBackgroundColor: colors.green1,
          focusBorderColor: colors.green1,
          activeBackgroundColor: colors.green1,
          activeBorderColor: colors.green1
        }
      }
    }
  }
);

const GreenTheme = ({ children, ...rest }) => (
  <ThemeProvider theme={config}>
    <StyledSpan>
      {React.Children.map(children, child => {
        if (React.isValidElement(child)) {
          return React.cloneElement(child, rest);
        }

        return child;
      })}
    </StyledSpan>
  </ThemeProvider>
);

GreenTheme.propTypes = {
  children: PropTypes.node.isRequired
};

GreenTheme.displayName = 'GreenTheme';

export default GreenTheme;
