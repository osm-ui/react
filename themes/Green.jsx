
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import {
    config as defaultThemeConfig,
    StyledDiv,
} from './Default';
import colors from '../colors';


export const config = _.merge({}, defaultThemeConfig, {
    color: colors.white,
    backgroundColor: colors.green2,
    borderColor: colors.green1,
    dimmerBackgroundColor: colors.green2,
    loaderColor: colors.white,
    controlColor: colors.green1,
    hoverControlColor: colors.white,

    form: {
        button: {
            color: colors.white,
            backgroundColor: colors.green2,
            borderColor: colors.green5,
            hoverBackgroundColor: colors.green3,
            hoverBorderColor: colors.green5,
            focusBackgroundColor: colors.green3,
            focusBorderColor: colors.green5,
            activeBackgroundColor: colors.green3,
            activeBorderColor: colors.green5,

            primary: {
                color: colors.green2,
                backgroundColor: colors.white,
                borderColor: colors.white,
                hoverBackgroundColor: colors.gray5,
                hoverBorderColor: colors.gray5,
                focusBackgroundColor: colors.gray5,
                focusBorderColor: colors.gray5,
                activeBackgroundColor: colors.gray5,
                activeBorderColor: colors.gray5,
            },

            success: {
                color: colors.white,
                backgroundColor: colors.green3,
                borderColor: colors.green3,
                hoverBackgroundColor: colors.green1,
                hoverBorderColor: colors.green1,
                focusBackgroundColor: colors.green1,
                focusBorderColor: colors.green1,
                activeBackgroundColor: colors.green1,
                activeBorderColor: colors.green1,
            },
        },
    },
});


const RedTheme = ({ children, ...rest }) => (
    <ThemeProvider theme={config}>
        <StyledDiv {...rest}>
            {children}
        </StyledDiv>
    </ThemeProvider>
);


RedTheme.propTypes = {
    children: PropTypes.node.isRequired,
};

RedTheme.displayName = 'RedTheme';

export default RedTheme;
