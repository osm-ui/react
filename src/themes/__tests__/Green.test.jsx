import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from '../../helpers/tests';
import Green from '../Green';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(Green));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(Green));
});
