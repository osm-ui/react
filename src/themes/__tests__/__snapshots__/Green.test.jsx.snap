// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<styled.span>
  <GreenTheme
    key=".0"
  >
    This is a text
  </GreenTheme>
</styled.span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<styled.span>
  <GreenTheme
    key=".0"
  >
    <div />
  </GreenTheme>
</styled.span>
`;
